src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANDOM = 30;
+    private static final int FIGURE_COUNT = 5;
+    private static final int DEFAULT = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, DEFAULT);
+    }
+
+    public Figure getRandomFigure() {
+        int randomize = random.nextInt(FIGURE_COUNT) + 1;
+        switch (randomize) {
+            case 1:
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(RANDOM));
+            case 2:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(RANDOM), random.nextInt(RANDOM));
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(), random.nextInt(RANDOM),
+                        random.nextInt(RANDOM));
+            case 4:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(RANDOM), random.nextInt(RANDOM), random.nextInt(RANDOM));
+            case 5:
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(RANDOM));
+            default:
+                return null;
```

Comment:
Best practice is to extract creating concrete figures to separate methods, for example `getRandomCircle`.

Reply:
It was like you say. Your colleague say to change it. What practice is the best?
https://github.com/mate-academy/jv-oop-advanced/pull/905#discussion_r1102790430